pkgs/by-name/to/todesk/package.nix

```diff
@@ -0,0 +1,147 @@
+{ stdenv
+, lib
+, fetchurl
+, dpkg
+, coreutils-full
+, autoPatchelfHook
+, makeWrapper
+, gzip
+, gnutar
+, nspr
+, kmod
+, systemdMinimal
+, glib
+, bubblewrap
+, libX11
+, libXrandr
+, glibc
+, libdrm
+, coreutils
+, libGL
+, bash
+, libXcomposite
+, libXdamage
+, libXfixes
+, libXtst
+, nss
+, libXxf86vm
+, gtk3
+, gdk-pixbuf
+, pango
+, appindicator-sharp
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "todesk";
+  version = "4.7.2.0";
+
+  src = fetchurl {
+    url = "https://newdl.todesk.com/linux/todesk-v${finalAttrs.version}-amd64.deb";
+    sha256 = "sha256-v7VpXXFVaKI99RpzUWfAc6eE7NHGJeFrNeUTbVuX+yg=";
+    curlOptsList = [ "--user-agent" "Mozilla/5.0" ];
+  };
+
+  nativeBuildInputs = [ dpkg autoPatchelfHook ];
+  buildInputs = [
+    nspr
+    kmod
+    systemdMinimal
+    glib
+    bash
+    coreutils
+    libX11
+    libXrandr
+    glibc
+    libdrm
+    libGL
+    libXcomposite
+    libXdamage
+    libXfixes
+    libXtst
+    nss
+    libXxf86vm
+    gtk3
+    gdk-pixbuf
+    pango
+    appindicator-sharp
+  ];
+
+  autoPatchelfIgnoreMissingDeps = [ "iHD_drv_video.so" "libglut.so" "libigdgmm.so" "libmfx.so" "libmfxhw64.so" "libva.so" "libva-drm.so" "libva-x11.so" "libzrtc.so" ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    dpkg -x $src ./todesk-src
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+        runHook preInstall
+        mkdir -p "$out"
+        cp -r todesk-src/* "$out"
+        echo "#! ${bash}/bin/bash -e" > "$out/opt/todesk/bin/ToDesk-Wrap"
+        echo "if [ ! -d /var/lib/todesk ]; then">> "$out/opt/todesk/bin/ToDesk-Wrap"
+        echo "mkdir /var/lib/todesk" >> "$out/opt/todesk/bin/ToDesk-Wrap"
+        echo "fi">> "$out/opt/todesk/bin/ToDesk-Wrap"
+        echo "${bubblewrap}/bin/bwrap --dev /dev --ro-bind /nix/store /nix/store --bind $out/opt/todesk/bin /opt/todesk/bin --ro-bind $out/opt/todesk/res /opt/todesk/res --bind /var/lib/todesk /opt/todesk/config --dev /dev \
```

Comment:
for better readbility consider either using a `let...in` construct, a `pkgs.writeShellScript` function or a `cat ... EOF` construct